### PR TITLE
dep: update tm-db to latest release

### PR DESCRIPTION
**Pull request by marbar3778**
Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

### CODE REVIEW

File: nodedb.go [hunk 1]
```diff
@@ -116,17 +116,22 @@ func (ndb *nodeDB) SaveNode(node *Node) {
...
 
 // Has checks if a hash exists in the database.
-func (ndb *nodeDB) Has(hash []byte) bool {
+func (ndb *nodeDB) Has(hash []byte) (bool, error) {
 	key := ndb.nodeKey(hash)
 
 	if ldb, ok := ndb.db.(*dbm.GoLevelDB); ok {
 		exists, err := ldb.DB().Has(key, nil)
 		if err != nil {
-			panic("Got error from leveldb: " + err.Error())
```
Comment (marbar3778) [inline nodedb.go]: do we want to keep the panic here? or let the end-user handle it? do you have an opinion @alexanderbez

Comment (alexanderbez) [reply nodedb.go]: It is of my opinion a DB library should never panic -- allow the upstream user to decide on how to handle the error.

File: nodedb.go [hunk 2]
```diff
@@ -73,8 +73,8 @@ func (ndb *nodeDB) GetNode(hash []byte) *Node {
 	}
 
 	// Doesn't exist, load.
-	buf := ndb.db.Get(ndb.nodeKey(hash))
-	if buf == nil {
+	buf, err := ndb.db.Get(ndb.nodeKey(hash))
+	if err != nil {
```
Comment (melekes) [inline nodedb.go]:
```suggestion
	if buf == nil {
```

File: nodedb.go [hunk 3]
```diff
@@ -73,8 +73,8 @@ func (ndb *nodeDB) GetNode(hash []byte) *Node {
...
 		panic(fmt.Sprintf("Value missing for hash %x corresponding to nodeKey %s", hash, ndb.nodeKey(hash)))
 	}
```
Comment (melekes) [inline nodedb.go]: I believe a new condition is needed

```
	if err != nil {
		panic(fmt.Sprintf("can't get node %X: %v", hash, err))
	}
```

File: nodedb.go (hunk 1 above)
Comment (melekes) [reply nodedb.go]: Strongly endorse ^

File: nodedb.go [hunk 4]
```diff
@@ -116,17 +119,22 @@ func (ndb *nodeDB) SaveNode(node *Node) {
...
 	key := ndb.nodeKey(hash)
 
 	if ldb, ok := ndb.db.(*dbm.GoLevelDB); ok {
 		exists, err := ldb.DB().Has(key, nil)
 		if err != nil {
-			panic("Got error from leveldb: " + err.Error())
+			return false, err
 		}
-		return exists
+		return exists, err
```
Comment (melekes) [inline nodedb.go]:
```suggestion
		return exists, nil
```

?

File: mutable_tree.go [hunk 5]
```diff
@@ -268,7 +268,10 @@ func (tree *MutableTree) LazyLoadVersion(targetVersion int64) (int64, error) {
 		targetVersion = latestVersion
 	}
 
-	rootHash := tree.ndb.getRoot(targetVersion)
+	rootHash, err := tree.ndb.getRoot(targetVersion)
+	if err != nil {
+		return 0, err
```
Comment (melekes) [inline mutable_tree.go]: so here is the question: at which point we should retry DB operation (say 5 times) before bubbling up an error to the caller?

Comment (marbar3778) [reply mutable_tree.go]: in my view, I would prefer the application to handle this and here we present the error each time. @alexanderbez do you have a preference. The sdk is the main user of this repo so your preference matters most

Comment (alexanderbez) [reply mutable_tree.go]: This change looks fine and behaves as I'd expect. If the version provided cannot be fetched, then the upstream caller should handle that error (either retry another version or fail completely).

Comment (melekes) [reply mutable_tree.go]: Agree. My question of course goes beyond this case.

**Pull request merged**